File: ios/Konnecta/UI/PlansHubViewModel.swift
import Foundation

@MainActor
class PlansHubViewModel: ObservableObject {
    private let activityService = ActivityService()
    
    @Published var activities: [ActivityWithParticipants] = []
    @Published var isLoading = false
    
    func loadFutureActivities(groupId: String) {
        isLoading = true
        Task {
            do {
                let dateFormatter = DateFormatter()
                dateFormatter.dateFormat = "yyyy-MM-dd"
                let today = dateFormatter.string(from: Date())
                
                let fetched = try await activityService.getFutureActivities(today: today, groupId: groupId)
                self.activities = fetched.sorted { ($0.start_date ?? $0.weekend_date) < ($1.start_date ?? $1.weekend_date) }
                isLoading = false
            } catch {
                print("Error loading future activities: \(error)")
                isLoading = false
            }
        }
    }
}

File: android/app/src/main/kotlin/com/konnecta/app/ui/viewmodel/PlansHubViewModel.kt
package com.konnecta.app.ui.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.konnecta.app.data.remote.ActivityService
import com.konnecta.app.data.remote.ActivityWithParticipants
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.launch
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class PlansHubViewModel : ViewModel() {
    private val activityService = ActivityService()

    private val _activities = MutableStateFlow<List<ActivityWithParticipants>>(emptyList())
    val activities: StateFlow<List<ActivityWithParticipants>> = _activities

    private val _isLoading = MutableStateFlow(false)
    val isLoading: StateFlow<Boolean> = _isLoading

    fun loadFutureActivities(groupId: String) {
        viewModelScope.launch {
            _isLoading.value = true
            try {
                val today = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault()).format(Date())
                val fetched = activityService.getFutureActivities(today, groupId)
                _activities.value = fetched.sortedBy { it.start_date ?: it.weekend_date }
            } catch (e: Exception) {
                // Handle error
            } finally {
                _isLoading.value = false
            }
        }
    }
}
